docs/arch/adr-010-new-versioning-schema.md

```diff
@@ -0,0 +1,119 @@
+# ADR 10: kURL new versioning scheme
+
+## Context
+Based on our prior experience, offering customers the ability to select various add-on versions provides them with significant flexibility, but it also creates numerous challenges. Presently, we're struggling to ensure that each customer's chosen add-ons are compatible with one another and that migration paths are available for all potential combinations. Moreover our user feedback indicates that stability and predictability are their top priorities, rather than the specific technologies they are running behind the scenes.
+
+## Decision
+Based on users feedback and the struggling we are facing to maintain kURL upgrades predictability, we have decided to adopt a more opinionated approach to delivering our Kubernetes distribution. This will involve implementing a well-designed versioning scheme, which is outlined in detail in this Architecture Decision Record.
+
+## Solution
+Under the new versioning scheme, our offering will be organised into Channels that correspond to specific Kubernetes minor versions. As of now, we will begin with a single channel named v1.27, which serves as our official offering for Kubernetes versions v1.27.x. Put simply, users who wish to install or keep their installations on Kubernetes v1.27 will remain on this Channel.
+
+Consequently, we can initiate the design of our new kURL installation YAML in the following manner:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27”
+```
+
+With the Channel (Kubernetes) now established as the basis for our delivery model, we can enable users to select a Version within this channel, which will allow them to access a specific set of Kubernetes-related features and functionalities (add-ons) that have been thoroughly tested and vetted by our team. This Version serves to abstract each individual add-on version into our own standardised version, ensuring that users can reliably and confidently utilise our offering without worrying about compatibility or interoperability issues.
+
+We will establish the practice of starting each Channel with a version v1.0.0 that groups together the initial set of add-ons (and respective versions) we support for that particular version. As a result, we will need to update the YAML file to reflect this approach, an initial Installer would look like this:
```

Comment:
Just to make sure we are on the same page here, the version 1.0.0 on Channel 1.26 would not be the same as the version 1.0.0 on Channel 1.27 (add-ons may differ on version and type, the logic may be different in the installer as may in the operator).

IIUC you are saying that we should be using, inside the Channels, "dates as versions" instead of semantic versions, if that is the case I would like to hear from others what they think .

/cc @rrpolanco @emosbaugh @kyleprice @camilamacedo86